[LRT][DOC] - Updating changelog in HIP up to ROCM7.12 (#2751)
ROCm/rocm-systems: projects/clr/CHANGELOG.md

Full documentation for HIP is available at [rocm.docs.amd.com](https://rocm.docs.amd.com/projects/HIP/en/latest/index.html)

## HIP 7.12 for ROCm 7.12

### Added

* New HIP APIs
    - Cooperative Groups

>       * Support for `barrier` APIs `barrier_arrive` and `barrier_wait` has been added for both `grid_group` and `thread_block`, enables finer‑grained synchronization within cooperative groups

Reviewer: ```suggestion
      * Support for `barrier` APIs `barrier_arrive` and `barrier_wait` has been added for both `grid_group` and `thread_block` to enable finer‑grained synchronization within cooperative groups
```

Author: Updated. Thanks.

      * Support for `block_rank` in the class `grid_group`, returns the rank of the block in the calling thread
    - Dynamic logging, no matching CUDA APIs exist
      * `hipExtEnableLogging` enables HIP runtime logging
      * `hipExtDisableLogging` disables HIP runtime logging
      * `hipExtSetLoggingParams` sets HIP runtime logging parameters

> * New HIP enumeration

Reviewer: ```suggestion

* New HIP enumeration
```

Author: Updated. Thanks.

    - `hipDeviceAttributeExpertSchedMode` has been added to hipDeviceAttribute_t to indicate whether expert scheduling mode is supported on AMD GPUs

### Resolved issues

* An error that occurred during HIP graph stream capture in thread‑local capture mode has been fixed. The HIP runtime now updates its validation logic to ensure that captures running in other threads on different streams no longer invalidate or block the thread‑local capture in the current thread.

### Optimized

> * HIP log level control capabilities HIP runtime adds dynamic logging functionalities, enabling applications to programmatically enable, disable, and configure logging at runtime without modifying environment variables or restarting the application. The result is more precise control over diagnostic output, making it easier to debug targeted code paths or minimize log noise during performance‑critical execution.

Reviewer: ```suggestion
* HIP log-level control capabilities HIP runtime adds dynamic logging functionalities, enabling applications to programmatically enable, disable, and configure logging at runtime without modifying environment variables or restarting the application. The result is more precise control over diagnostic output, making it easier to debug targeted code paths or minimize log noise during performance‑critical execution.
```

Author: Updated. Thanks.
